src/Tokenizer/Analyzer/Analysis/DataProviderAnalysis.php

```diff
@@ -14,20 +14,30 @@
 
 namespace PhpCsFixer\Tokenizer\Analyzer\Analysis;
 
+use PhpCsFixer\Console\Application;
+use PhpCsFixer\Utils;
+
 final class DataProviderAnalysis
 {
     private string $name;
 
     private int $nameIndex;
 
-    /** @var array<int> */
+    /** @var list<int> */
     private array $usageIndices;
 
     /**
-     * @param array<int> $usageIndices
+     * @param list<int> $usageIndices
      */
     public function __construct(string $name, int $nameIndex, array $usageIndices)
     {
+        if (!array_is_list($usageIndices)) {
+            Utils::triggerDeprecation(new \InvalidArgumentException(sprintf(
+                'Parameter "usageIndices" should be list (and not array) and this will be enforced in version %d.0.',
```

Comment:
I find the "(and not array)" part confusing, maybe describe what list is instead, or just drop this?

Reply:
`list` and `array` are established types, we do not describe the difference between them anywhere (and we use in lot of places)

Reply:
```suggestion
                'Parameter "usageIndices" should be of type 'list' (and not type 'array') and this will be enforced in version %d.0.',
```